docs/docs/adding-markdown-pages.md

```diff
@@ -2,7 +2,162 @@
+node api, which help you create static pages automatically.
+Here's how you do it step by step.
+
+1. Read files in Gatsby from the filesystem.
+2. Understand Markdown syntax. Convert metadata to api and content to html.
+3. Create a page template from above Markdown data and style it.
+4. Create static pages using Gatsby's node api at paths you like.
+
+### Read files in Gatsby from the filesystem - `gatsby-source-filesystem`
+
+Use the plugin [`gatsby-source-filesystem`](https://www.gatsbyjs.org/packages/gatsby-source-filesystem/#gatsby-source-filesystem) to read files off of defined file paths.
+
+`npm i --save gatsby-source-filesystem`
+
+Now open up your `gatsby-config.js` to add this plugin to the plugin list.
+
+The plugin list accepts either a string which is the plugin name or an object with any options you may want to pass.
+Here we add an object with the path as an option to the plugin
+
+```
+plugins: [
+    `...`,
+    {
+      resolve: `gatsby-source-filesystem`,
+      options: {
+            path: `${__dirname}/path/for/markdown/files`,
+            name: 'markdownpages'
+        }
+    }
+]
+```
+The output of this plugin can be consumed by 'transformer' plugins in Gatsby.
+
+### Understand markdown syntax. - `gatsby-transformer-remark`
+
+Use the plugin [`gatsby-transformer-remark`](https://www.gatsbyjs.org/packages/gatsby-transformer-remark/) to recognise files which are markdown and read it's content. It will convert the frontmatter metadata part of your markdown file as `frontmatter` and the content part as html.
+
+`npm i --save gatsby-transformer-remark`
+
+Add this to `gatsby-config.js`. Since there are no options needed,
+
+```
+ plugins: [`gatsby-transformer-remark`]
+```
+
+#### Note on creating markdown files.
+
+When you create a Markdown file, you can add a block that looks like below to the top of the file content. This block will be parsed by `gatsby-transformer-remark` as `frontmatter`. The GraphQL api will provide this data in our React components.
+
+```
+---
+path: "/blog/my-first-post"
+date: "2017-11-07"
+title: My first blog post"
+---
+```
+
+### Create a page template for the markdown data.
+
+Create a folder in the `/src` directory of your Gatsby application called `templates`.
+Now create a `blogTemplate.js` inside it with the following content.
+
+```
+import React from "react"
+
+export default function Template({
+  data, // this prop will be injected by the GraphQL query below.
+}) {
+  const { markdownRemark } = data // data.markdownRemark holds our post data
+  const {frontMatter, html} = markdownRemark;
+  return (
+    <div className="blog-post-container">
+      <div className="blog-post">
+        <h1>{frontmatter.title}</h1>
+        <h2>{frontmatter.date}</h2>
+        <div
+          className="blog-post-content"
+          dangerouslySetInnerHTML={{ __html: html }}
+        />
+      </div>
+    </div>
+  )
+}
+
+export const pageQuery = graphql`
+  query BlogPostByPath($path: String!) {
```

Comment:
Could you offer just a little more about who / how the `$path` variable is getting filled in? I thought it would get set in the  `context` parameter in `gatsby-node.js` but I don't see it there.

Reply:
Thanks for the review!
So that's a graphQL thingy which I don't completely understand honestly :( 
Regardless, wouldn't the right thing to do be point to graphQL docs or graphQL-Gatsby docs at this stage? It would be awfully irrelavant explaining graphQL query structure here no?

Reply:
Sorry, I don't mean explain how graphql variables work, I mean specifically what sets the value for that `$path` variable. If that's a standard thing that's explained somewhere else, then a pointer would be good. I haven't found a place it's explained, though. I only figured out some of how they work by reading code in the examples (gatsbygram in particular).

Reply:
Also, I really want to say thank you for writing this, it's already helped me a lot.

Reply:
I'm sorry for the delay. I really am only trying this out on the weekend. Thanks for the encouragement. Will finish this up soon.

Reply:
@tamouse - I tried figuring this out. I think I have an idea that possibly some Gatsby Node-api magic is involved, but can't say for sure. Would really love somebody's help on this. Maybe @KyleAMathews can help me with this part of the documentation.

Reply:
@tayiorbeii talks about this in https://egghead.io/lessons/gatsby-create-a-blog-post-template-for-gatsby .Maybe he can help?